From d7b4033c08ba4e5c9c7ddd0a7b62120056fd5f40 Mon Sep 17 00:00:00 2001
From: Ira W. Snyder <iws@ovro.caltech.edu>
Date: Mon, 7 Dec 2009 23:21:28 -0500
Subject: [PATCH 105/119] vbus: kvm connector: fix most sparse warnings

This fixes a few warnings from sparse. There is still one warning
remaining, but I am unsure how to fix it:

kernel/vbus/connectors/kvm.c:985:28: error: bad constant expression

Signed-off-by: Ira W. Snyder <iws@ovro.caltech.edu>
Signed-off-by: Gregory Haskins <ghaskins@novell.com>
---
 kernel/vbus/connectors/kvm.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/vbus/connectors/kvm.c b/kernel/vbus/connectors/kvm.c
index 54e650f..02111e1 100644
--- a/kernel/vbus/connectors/kvm.c
+++ b/kernel/vbus/connectors/kvm.c
@@ -38,7 +38,7 @@ struct _eventq {
 		struct kvm_xvmap *xvmap;
 	} ringdata;
 	struct work_struct  wakeup;
-	int                 backpressure:1;
+	bool                backpressure;
 };
 
 enum _state {
@@ -231,7 +231,7 @@ static struct vbus_memctx_ops _memctx_ops = {
 	.release   = &_memctx_release,
 };
 
-struct vbus_memctx *_memctx_alloc(struct vbus_kvm *vkvm)
+static struct vbus_memctx *_memctx_alloc(struct vbus_kvm *vkvm)
 {
 	struct _memctx *_memctx;
 
@@ -399,7 +399,7 @@ events_flush(struct _eventq *eventq)
 		if (iter.desc->len < sizeof(*ev)) {
 			SHM_SIGNAL_FAULT(eventq->ioq->signal,
 					 "Desc too small on eventq: " \
-					 "%p: %ld<%ld",
+					 "%p: %ld<%zu",
 					 (void *)iter.desc->ptr,
 					 (unsigned long)iter.desc->len,
 					 sizeof(*ev));
-- 
1.6.5.2

